rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/Sw360AuthenticationProvider.java

```diff
@@ -47,59 +59,100 @@
+        return authentication.equals(UsernamePasswordAuthenticationToken.class);
+    }
+
+    private boolean isAuthorized(String email, String password) {
+        // Solution 1:
+        // UserLocalServiceUtil.authenticateForBasic
+        // userId = UserLocalServiceUtil.authenticateForBasic(companyId, authType, login, current);
+        // this need a dependency to liferay
+
+        // Solution 2:
+        // Liferay json webservice call to verify username and password
+        String liferayParameterURL = "/api/jsonws/user/get-user-id-by-email-address?companyId=%s&emailAddress=%s";
+        String url = sw360PortalServerURL + String.format(liferayParameterURL, sw360LiferayCompanyId, email);
+        RestTemplateBuilder restTemplateBuilder = new RestTemplateBuilder();
+        String encodedPassword;
+        try {
+            encodedPassword = URLDecoder.decode(password, "US-ASCII");
+        } catch (UnsupportedEncodingException e) {
+            return false;
+        }
+        RestTemplate restTemplate = restTemplateBuilder.basicAuthorization(email, encodedPassword).build();
+        ResponseEntity<String> response = restTemplate.postForEntity(url, null, String.class);
+        return (parseInteger(response.getBody()) > 0);
+    }
+
+    private User getUserByEmail(String email) {
+        UserService.Iface client = new ThriftClients().makeUserClient();
+        User user = null;
+        try {
+            if (!Strings.isNullOrEmpty(email) && client != null) {
+                user = client.getByEmail(email);
+            }
+        } catch (TException e) {
+            user = null;
+        }
+        return user;
+    }
+
+    private Authentication createAuthenticationToken(String name, String password, User user) {
         List<GrantedAuthority> grantedAuthorities = new ArrayList<>();
-        grantedAuthorities.add(new SimpleGrantedAuthority("ROLE_SW360_USER"));
+        grantedAuthorities.add(new SimpleGrantedAuthority(READ.getAuthority()));
+        if (!Objects.isNull(user) && PermissionUtils.isUserAtLeast(WRITE_ACCESS_USERGROUP, user)) {
+            grantedAuthorities.add(new SimpleGrantedAuthority(WRITE.getAuthority()));
+        }
         return new UsernamePasswordAuthenticationToken(name, password, grantedAuthorities);
     }
 
-    @Override
-    public boolean supports(Class<?> authentication) {
-        return authentication.equals(UsernamePasswordAuthenticationToken.class);
+    private boolean isDevEnvironment() {
+        Boolean result = false;
+        String[] activeProfiles = environment.getActiveProfiles();
+        for (String profile : activeProfiles) {
+            if (profile.equals(ENVIRONMENT_DEV_PROFILE)) {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    private int parseInteger(String value) {
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            return 0;
```

Comment:
Corrected. I did a mistake in the refactoring.
I removed the parse integer method. Instead this method I only try to parse the integer value.
The user exits in liferay if the response body contains a valid number. Otherwise the user is not authorized.